docs/streams/developer-guide.html

```diff
@@ -267,21 +295,532 @@
+    stateless and stateful transformations.
+
+   
+
+    <h5><a id="streams_dsl_transformations_stateless" href="#streams_dsl_transformations_stateless">Stateless transformations</a></h5>
+    Stateless transformations, by definition, do not depend on any state for processing, and hence implementation-wise they do not
+    require a state store associated with the stream processor.
+
+
+<table class="data-table" border="1">
+    <tbody><tr>
+        <th>Transformation</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td><b>Branch</b>: <code>KStream->KStream<</code></td>
+        <td>
+            Branch (or split) a <code>KStream</code> based on the supplied predicates into one or more <code>KStream</code> instances.<p>
+
+            Predicates are evaluated in order. A record is placed to one and only one output stream on the first match:
+            if the n-th predicate evaluates to true, the record is placed to n-th stream. If no predicate matches,
+            the record is dropped.<p>
+
+            Branching is useful, for example, to route records to different downstream topics.
+
+            <pre class="brush: java;">
+            KStream&lt;String, Long&gt; stream = ...;
+            KStream&lt;String, Long&gt;[] branches = stream.branch(
+		            (key, value) -> key.startsWith("A"), /* first predicate  */
+		            (key, value) -> key.startsWith("B"), /* second predicate */
+		            (key, value) -> true                 /* third predicate  */
+            );
+            // KStream branches[0] contains all records whose keys start with "A"
+            // KStream branches[1] contains all records whose keys start with "B"
+            // KStream branches[2] contains all other records
+            // Java 7 example: cf. `filter` for how to create `Predicate` instances
+	        </pre>
+        </td>
+    </tr>
+	 <tr>
+	   <td><b>Filter</b>: <code>KStream->KStream or KTable->KTable</code></td>
+	   <td>
+	     Evaluates a boolean function for each element and retains those for which the function returns true. 
+
+	     <pre class="brush: java;">
+             KStream&lt;String, Long&gt; stream = ...;
+
+             // A filter that selects (keeps) only positive numbers
+             // Java 8+ example, using lambda expressions
+             KStream&lt;String, Long&gt; onlyPositives = stream.filter((key, value) -> value > 0);
+
+             // Java 7 example
+             KStream&lt;String, Long&gt; onlyPositives = stream.filter(
+               new Predicate&lt;String, Long&gt;() {
+                 @Override
+                 public boolean test(String key, Long value) {
+                   return value > 0;
+                 }
+               });
+	     </pre>
+	   </td>
+	 </tr>
+	 <tr>
+	   <td><b>Inverse Filter</b>: <code>KStream->KStream or KTable->KTable</code></td>
+	   <td>
+	     Evaluates a boolean function for each element and drops those for which the function returns true.
```

Comment:
Ping @derrickdoo: With the current design/style, this table is not rendered correctly.

Screenshot:

<img width="1070" alt="screen shot 2017-07-17 at 17 25 41" src="https://user-images.githubusercontent.com/294849/28275697-fb4e933c-6b14-11e7-8dce-ba05d54cc455.png">

Note how e.g. the first line stops at "including t".  You can manually "scroll" by highlighting the text with your mouse and then dragging your mouse to the right while keeping the left button pressed, but that ain't cool. ;-)